### PR TITLE
Ken/uvminfo

### PR DESCRIPTION
**Pull request by KenGordon**
Use the environment provided uvm reference info (cosesign1 wrapped json containing the uvm launch measurement) and amd certificates. Update to latest MAA to match.

### CODE REVIEW

File: pkg/attest/maa.go [hunk 1]
```diff
@@ -50,33 +60,56 @@ type attestSNPRequestBody struct {
 // and the certificate chain, (ii) the  runtime data (base64 URL encoding of the public
 // wrapping key), (iii) the inittime data (base64 URL encoding of the security policy),
 // and (iv) a nonce
-func newAttestSNPRequestBody(snpAttestationReport, vcekCertChain, policyBlob, keyBlob []byte) (*attestSNPRequestBody, error) {
+func newAttestSNPRequestBody(snpAttestationReport []byte, vcekCertChain []byte, policyBlob []byte, keyBlob []byte, uvmReferenceInfo []byte) (*attestSNPRequestBody, error) {
```
Comment (svolos) [inline pkg/attest/maa.go:63]: Can we just remove the commented out lines?

File: pkg/attest/maa.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+
+	// "io/ioutil" needed when generating test data
```
Comment (svolos) [inline pkg/attest/maa.go]: can we please remove commented out lines? or use a flag to decide whether to print or not

File: pkg/attest/data/uvm_host_amd_certificate.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (svolos) [inline pkg/attest/data/uvm_host_amd_certificate.json]: Are these test data used by attest_test ? It may make sense to rename data to test_data ?

File: pkg/attest/certcache.go [hunk 4]
```diff
@@ -3,6 +3,13 @@
 
 package attest
 
+/*
```
Comment (svolos) [inline pkg/attest/certcache.go:6]: I think we may need to keep this around, as it is the only way to request certificates from the AMD cert cache. But we can leave this for another PR where we revisit this.

File: pkg/attest/attest.go [hunk 5]
```diff
@@ -53,11 +58,26 @@ func Attest(certCache CertCache, maa MAA, inittimeDataBytes []byte, runtimeDataB
...
+		return "", errors.Wrap(err, "decoding policy from Base64 format failed")
+	}
+	logrus.Debugf("   inittimeDataBytes:    %v", inittimeDataBytes)
+
 	SNPReportBytes, err := FetchSNPReport(fetchRealSNPReport, runtimeDataBytes, inittimeDataBytes)
 	if err != nil {
 		return "", errors.Wrapf(err, "fetching snp report failed")
 	}
 
+	/*
```
Comment (svolos) [inline pkg/attest/attest.go:72]: I think that's where we should be using the AMD Cert cache endpoint just in case we need to fetch fresh certificates.

File: docker/skr/Makefile [hunk 6]
```diff
@@ -0,0 +1,83 @@
+#
```
Comment (svolos) [inline docker/skr/Makefile]: why is this file part of the PR ?

File: cmd/skr/main.go [hunk 7]
```diff
@@ -244,6 +232,16 @@ func main() {
 	logFile := flag.String("logfile", "", "Logging Target: An optional file name/path. Omit for console output.")
 	port := flag.String("port", "8080", "Port on which to listen")
 
+	// WARNING!!!
+	// If the security policy does not control the areguments to this process then
```
Comment (svolos) [inline cmd/skr/main.go]: areguments -> arguments

**Review by svolos — COMMENTED** [review]
I have talked with Ken offline and agreed with the changes to the APIs to pass the uvm information and dropping the certcache info. Only few points to consider:
- Change the data directory to test_data 
- Better way to produce the test_data as opposed to commented out lines
- Consider utilizing a default certcache endpoint to fetch fresh certificates. [This can be done via a new PR]

File: docker/skr/Makefile (hunk 6 above)
Comment (KenGordon) [reply docker/skr/Makefile]: Removed

File: pkg/attest/data/uvm_host_amd_certificate.json (hunk 3 above)
Comment (KenGordon) [reply pkg/attest/data/uvm_host_amd_certificate.json]: done

File: cmd/skr/main.go (hunk 7 above)
Comment (KenGordon) [reply cmd/skr/main.go]: done

File: pkg/attest/maa.go (hunk 2 above)
Comment (KenGordon) [reply pkg/attest/maa.go]: Now in common there is a flag "GenerateTestData" that controls this.

File: pkg/attest/certcache.go (hunk 4 above)
Comment (KenGordon) [reply pkg/attest/certcache.go:6]: I have updated the comment.

**Review by svolos — APPROVED** [review]
